File: docs/examples/go/list_whitelist.go
// Command list_whitelist fetches every whitelist record.
package main

import (
    "encoding/json"
    "fmt"
    "net/http"
)

const whitelistToken = "<token>"

func main() {
    req, _ := http.NewRequest(http.MethodGet, baseURL+"/settings/whitelist", nil)
    req.Header.Set("Authorization", "Bearer "+whitelistToken)

    resp, err := http.DefaultClient.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        panic(fmt.Sprintf("unexpected status: %d", resp.StatusCode))
    }

    var records []struct {
        IP          string `json:"ip"`
        Description string `json:"description"`
    }
    if err := json.NewDecoder(resp.Body).Decode(&records); err != nil {
        panic(err)
    }

    for _, record := range records {
        desc := record.Description
        if desc == "" {
            desc = "no description"
        }
        fmt.Printf("%s -> %s\n", record.IP, desc)
    }
}

File: docs/examples/python/list_whitelist.py
"""Retrieve the whitelist contents."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"


def list_whitelist() -> None:
    response = requests.get(
        f"{BASE_URL}/settings/whitelist",
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    for item in response.json():
        print(f"{item['ip']} ({item.get('description', 'no description')})")


if __name__ == "__main__":
    list_whitelist()
